components/hab/src/cli/hab/sup.rs

```diff
@@ -338,6 +338,21 @@ pub struct SupRun {
     /// process (default: set in plan)
     #[structopt(name = "SHUTDOWN_TIMEOUT", long = "shutdown-timeout")]
     shutdown_timeout: Option<ShutdownTimeout>,
+    /// Automatically cleanup old package versions.
+    ///
+    /// If this flag is enabled, service startup will initiate an uninstall of all previous
+    /// versions of the associated package. This also applies when a service is restarted due to an
+    /// update. If a number is passed to this argument, that number of latest versions will be
+    /// kept. The same logic applies to the Supervisor package if self updates are enabled.
+    // TODO (DM): We want this to have a default value of 1 if set otherwise it should be none. Does
+    // Option<Option<T>> achieve this
+    #[structopt(name = "KEEP_LATEST_PACKAGES",
+                long = "keep-latest-packages",
+                default_value = "1",
+                max_values = 2, // This is needed to use an env var
+                env = "HAB_KEEP_LATEST_PACKAGES")]
+    #[allow(clippy::option_option)]
+    keep_latest_packages: Option<Option<usize>>,
```

Comment:
I think this is the first time I've ever seen an optional Option :rofl:

Since this is weird enough to have its own Clippy lint, I wonder if we could create a custom type that better encapsulates the behavior we're after here... :thinking: 

(On further reading, I'm not 100% clear what the nested `Option` indicates... it seems like we either have `None`, or we have `Some(non-zero number)`. Is it required in this form to play nicely with Clap somehow?)

Reply:
Yeah, that was a mess. That is some `structopt` [type magic](https://docs.rs/structopt/0.3.12/structopt/index.html#type-magic) to try and indicate that an optional argument can optionally take a value. In this case, I wanted`--keep-latest-packages` to default to `1` if no explicit value was specified.

There are a few issues that keep this from working well ([here](https://github.com/TeXitoi/structopt/issues/366) and [here](https://github.com/TeXitoi/structopt/issues/368)). In [1036bc](https://github.com/habitat-sh/habitat/pull/7582/commits/1036bc25017ea5df6a92a7bd6e8537e4ce175b03) I switched to using a normal `Option` which requires the user explicitly set `--keep-latest-packages=1` instead of just `--keep-latest-packages`.